Sources/Common/Communication/EventStorage.swift

```diff
@@ -1,90 +1,69 @@
 import Foundation
 
 public protocol EventStorage {
-    func store<E: Codable>(
-        // sourcery:Type=AnyEncodable
-        // sourcery:TypeCast="AnyEncodable(event)"
-        event: E,
-        forKey key: String
-    ) throws
-    func loadAllEvents<E: EventRepresentable>(
-        ofType type: E.Type,
-        withKey key: String
-    ) throws -> [E]
-    func clearEvent(forKey key: String) throws
-    func clearAllEvents() throws
+    /// Stores an event.
+    func store<E: EventRepresentable & Codable>(event: E) throws
+    /// Loads all events of a specific type.
+    func loadEvents<E: EventRepresentable & Codable>(ofType type: E.Type) throws -> [E]
+    /// Removes a specific event by its storage identifier.
+    func remove<E: EventRepresentable>(ofType eventType: E, withStorageId storageId: String) throws
 }
 
 // sourcery: InjectRegisterShared = "EventStorage"
 public class EventStorageManager: EventStorage {
-    private let fileManager: FileManager = .default
-    private let documentsDirectory: URL
+    private let fileManager = FileManager.default
+    private let baseDirectory: URL
+    private let encoder = JSONEncoder()
+    private let decoder = JSONDecoder()
+    private let ioQueue: DispatchQueue
```

Comment:
I was expecting to see dependencies such as `JsonAdapter`, `FileStorage`, and `ThreadUtil/Lock/LockManager` used here. 

Was there a reason to avoid using them here?

Reply:
I was unsure about future of `JsonAdapter` as it has alot of code added for BQ, if all of it goes away, would we still keep it? 

`FileStorage` simialrly, is oriented around BQ, example below, 

```
public enum FileType {
    case queueInventory
    case queueTask
    }
    func save(type: FileType, contents: Data, fileId: String?) -> Bool
```

Reply:
Good question. 

The `FileStorage` implementation could use a good update. It's quite bias towards the BQ at the moment. The `JsonAdapter`, I am not sure if we need to make any changes to it to be useful outside of the BQ. 

These classes are heavily used in the BQ today. I do however still see a lot of value in using these classes. Reducing boilerplate code, encapsulated error handling, and easy mocking to name a few reasons. 

Therefore, I do think it would be wise to continue using these classes. Even if we decide to change some of the implementation (such as the File storage)